quodlibet/quodlibet/formats/_audio.py

```diff
@@ -536,24 +537,77 @@ def _role_call(self, role_tag, sub_keys=None):
 
     @property
     def lyric_filename(self):
-        """Returns the (potential) lyrics filename for this file"""
-
-        filename = self.comma("title").replace(u'/', u'')[:128] + u'.lyric'
-        sub_dir = ((self.comma("lyricist") or self.comma("artist"))
-                  .replace(u'/', u'')[:128])
-
-        if os.name == "nt":
-            # this was added at a later point. only use escape_filename here
-            # to keep the linux case the same as before
-            filename = escape_filename(filename)
-            sub_dir = escape_filename(sub_dir)
-        else:
-            filename = fsnative(filename)
-            sub_dir = fsnative(sub_dir)
-
-        path = os.path.join(
-            expanduser(fsnative(u"~/.lyrics")), sub_dir, filename)
-        return path
+        """Returns the validated, or default, lyrics filename for this
+        file. User defined '[memory] lyric_rootpaths' and
+        '[memory] lyric_filenames' matches take precedent"""
+
+        from quodlibet.pattern import ArbitraryExtensionFileFromPattern
+
+        lyric_paths = \
+            config.getstringlist("memory", "lyric_rootpaths", [])
+        # add default
+        lyric_paths.append(u'~/.lyrics')
+        lyric_filenames = \
+            config.getstringlist("memory", "lyric_filenames", [])
+        # add defaults
+        lyric_filenames.append(
+            (self.comma("lyricist") or self.comma("artist"))
+            .replace(u'/', u'')[:128] + os.sep +
```

Comment:
Can we extract this repeated string munging to a named function please (e.g. `sanitise`, or something)

Reply:
good call. done